account/verification_code.go

```diff
@@ -0,0 +1,207 @@
+package account
+
+import (
+	"context"
+	"strconv"
+	"time"
+
+	"github.com/fabric8-services/fabric8-auth/application/repository"
+	"github.com/fabric8-services/fabric8-auth/errors"
+	"github.com/fabric8-services/fabric8-auth/gormsupport"
+	"github.com/fabric8-services/fabric8-auth/log"
+
+	"github.com/goadesign/goa"
+	"github.com/jinzhu/gorm"
+	errs "github.com/pkg/errors"
+	uuid "github.com/satori/go.uuid"
+)
+
+type VerificationCode struct {
+	gormsupport.Lifecycle
+	ID       uuid.UUID `sql:"type:uuid default uuid_generate_v4()" gorm:"primary_key"` // This is the ID PK field
+	User     User
+	Verified bool // The verification status of the updated email.
+	Code     string
+}
+
+// TableName overrides the table name settings in Gorm to force a specific table name
+// in the database.
+func (m VerificationCode) TableName() string {
+	return "verification_codes"
+}
+
+// GetETagData returns the field values to use to generate the ETag
+func (m VerificationCode) GetETagData() []interface{} {
+	// using the 'ID' and 'UpdatedAt' (converted to number of seconds since epoch) fields
+	return []interface{}{m.ID, strconv.FormatInt(m.UpdatedAt.Unix(), 10)}
+}
+
+// GetLastModified returns the last modification time
+func (m VerificationCode) GetLastModified() time.Time {
+	return m.UpdatedAt
+}
+
+// GormVerificationCodeRepository is the implementation of the storage interface for
+// VerificationCode.
+type GormVerificationCodeRepository struct {
+	db *gorm.DB
+}
+
+// NewVerificationCodeRepository creates a new storage type.
+func NewVerificationCodeRepository(db *gorm.DB) *GormVerificationCodeRepository {
+	return &GormVerificationCodeRepository{db: db}
+}
+
+// VerificationCodeRepository represents the storage interface.
+type VerificationCodeRepository interface {
+	repository.Exister
+	Load(ctx context.Context, id uuid.UUID) (*VerificationCode, error)
+	LoadByCode(ctx context.Context, code string) (*VerificationCode, error)
+	Create(ctx context.Context, VerificationCode *VerificationCode) error
+	Save(ctx context.Context, VerificationCode *VerificationCode) error
+	Delete(ctx context.Context, id uuid.UUID) error
+	Query(funcs ...func(*gorm.DB) *gorm.DB) ([]VerificationCode, error)
```

Comment:
No need for Query. Create(), Load() and Delete() I thin is the only methods we need.

Reply:
Had kept Query() for using with Filter..() .. in case we ever wanted to audit 'how many times did a user update his email and when'

Reply:
We can always add it to the code if/when we need it, right? Don't want to add blow up our code when not necessary. I would expect we will need to audit the DB directly if we need it. Query&Filters are useless unless we add some secured API or even GUI to it.